crates/wasmtime/src/trampoline/func.rs

```diff
@@ -261,7 +309,35 @@ pub fn create_handle_with_function(
     .map(|instance| (instance, trampoline))
 }
 
-pub unsafe fn create_handle_with_raw_function(
+pub unsafe fn create_raw_engine_function(
+    func: *mut [VMFunctionBody],
+    host_state: Box<dyn Any>,
```

Comment:
Will add.

Reply:
So there's a problem with adding this bound.

We're passing `self` in the `IntoFunc` impls, which has type `F` with bounds `F: Fn(Caller<'_>, $($args),*) -> R + 'static` and unfortunately that bound is shared with `into_func` which wouldn't want those bounds on `F` otherwise it pollutes `Func::wrap`.

To properly implement this, we'd need a separate `IntoEngineFunc` impl.  Thoughts?

Reply:
Or would we rather add these bounds here and transmute them in for the `IntoFunc` impl with a big comment as to why this is safe (i.e. because the bounds are present in the `Engine::wrap_func`)?